lib/Backend/LowerMDShared.cpp

```diff
@@ -9321,6 +9332,39 @@ void LowererMD::GenerateFastInlineBuiltInMathAbs(IR::Instr* inlineInstr)
     }
 }
 
+void LowererMD::GenerateFastInlineBuiltInMathPow(IR::Instr* instr)
+{
+#ifdef _M_IX86
+    AssertMsg(!instr->GetSrc2()->IsFloat(), "Math.pow(*, double) needs customized lowering!");
+#endif
+
+    IR::JnHelperMethod directPowHelper = (IR::JnHelperMethod)0;
+
+    if (instr->GetSrc2()->IsFloat())
+    {
+        AssertMsg(instr->GetSrc1()->IsFloat(), "Math.Pow(int, double) should not generated by GlobOpt!");
+        directPowHelper = IR::HelperDirectMath_Pow;
```

Comment:
This looks good. Will enclose instr->GetSrc2()->IsFloat() into ifdef.

---
In reply to: [57633793](https://github.com/Microsoft/ChakraCore/pull/644#discussion_r57633793) [](ancestors = 57633793)